source/Calamari.Common/Features/Packages/PackageName.cs

```diff
@@ -175,6 +175,20 @@ public static PackageFileNameMetadata FromFile(string? path)
             return new PackageFileNameMetadata(packageId, version, extension);
         }
 
+        public static bool TryFromFile(string? path, out PackageFileNameMetadata? metadata)
+        {
+            try
+            {
+                metadata = FromFile(path);
+                return true;
+            }
+            catch
+            {
+                metadata = new PackageFileNameMetadata("", new SemanticVersion(0,0,0) ,"");
```

Comment:
Why did you opt to set to the metadata in the case of an exception?

Reply:
No idea.  It should be `metadata = default;`